service/service.go

```diff
@@ -107,21 +112,54 @@ func (s *Service) CreateUser(ctx context.Context, req *pb.UserRequest) (*pb.User
 	user.Password = hashedPassword
 	user.IsVerified = false
 
-	// insert into DB
+	// create unique email token
+	token, err := generateEmailToken()
```

Comment:
What if users use similar e-mails?

Reply:
I will move this block after I insert the new user.
Duplicate email will be catched by the db as unique violation error, and will return the duplicate email error.